content/v2.0/router/basic-usage.md

```diff
@@ -0,0 +1,267 @@
+
+```ruby
+# /slices/main/lib/main/views/home/index.rb
+
+module Main
+  module Views
+    module Home
+      class Index < Main::View
+      end
+    end
+  end
+end
+```
+
+```ruby
+# /slices/main/lib/main/templates/home/index.html.slim
+
+h1 Hello, Hanami!
+```
+
+This is great, as it allows us to write our templates in the `HTML` or `Slim` files, or serialize our `JSON` responses using serializers instead of using raw strings everywhere!
+
+Read more about [Hanami::Actions](/v2.0/actions/overview) and [rendering views](/v2.0/views/overview)!.
+
+# Features
+
+`Hanami::Router` comes with a full set of features helping to organize your routes, no matter how big or complex your application is. It's designed to scale!
+
+## Slices
+
+One of the unique Hanami features is how it encourages modularizing your application into multiple, self-contained building blocks. We call it [slices](/v2.0/slices/overview).
+
+A slice is an independent piece of application. Think of it as different entry points to your domain.
+
+```ruby
+# /config/routes.rb
+
+Hanami.application.routes do
+  slice :main, at: "/" do
+    root to: "home.show"  # slices/main/lib/actions/home/show.rb
+    get "/hello", to: "hello.index" # slices/main/lib/actions/hello/index.rb
+  end
+
+  slice :admin, at: "/" do
+    root to: "dashboard.show" # slices/admin/lib/actions/dashboard/show.rb
+    get "/users", to: "users.index" # slices/admin/lib/actions/users/index.rb
+  end
+end
+```
+
+Read more about [slices](/v2.0/slices/overview)!
+
+## URLs
+
+### Path matching
+
+In [our initial example](/v2.0/router/overview) we have introduced a really basic relative URI: `/hello`.
+This is what we call _fixed path matching_.
+It is called this because the segment is responsible for responding only to an **exact match**.
+If we visit `/hello`, we get a response.
+If we hit `/foo`, a `404` (Not Found) is returned.
+
+### Fixed Matching
+
+```ruby
+# config/routes.rb
+
+Hanami.application.routes do
+  get "/dashboard", to: "dashboard.index"
+end
+```
+
+### Variables
+
+When we have dynamic content to serve, we want our URI to be dynamic as well.
+This can be easily achieved via path variables.
+They are defined with a colon, followed by a name (eg. `:id`).
+
+Once an incoming request is forwarded to our endpoint, we can access the current value in our param"s action (`params[:id]`).
+
+```ruby
+# /config/routes.rb
+
+Hanami.application.routes do
+  get "/books/:id", to: "books.show"
+end
+```
+
+Multiple variables can be used in a path.
+
+```ruby
+# /config/routes.rb
+
+Hanami.application.routes do
+  get "/books/:book_id/reviews/:id", to: "reviews.show"
+end
+```
+
+### Variables Constraints
+
+It"s possible to specify constraints for each variable.
+The rule MUST be expressed as a regular expression.
+If a request can satisfy all of them, we"re good, otherwise, a `404` is returned.
```

Comment:
```suggestion
If a request can satisfy all of them, we're good, otherwise, a `404` is returned.
```

Reply:
Ah, I guess you replaced `'` with `"` throughout, I see now :)

Reply:
Oh, yeah I've got used to single quotes in the code, but hanami developers prefer double quotes so I wanted to unify to keep things tied together :). Too quick. Will look for the rest of overlooks, thanks for the review!